src/services/positionService.ts

```diff
@@ -704,37 +704,34 @@ export default class PositionService {
 
     const hasYield = position.to.underlyingTokens.length;
 
-    if (!useProtocolToken && !hasYield) {
-      const hubInstance = await this.contractService.getHubInstance(position.version);
+    const companionHasPermission = await this.companionHasPermission(position, DCAPermission.WITHDRAW);
 
-      return hubInstance.withdrawSwapped(position.positionId, this.walletService.getAccount());
-    }
+    let permissionPermit: Awaited<ReturnType<typeof this.getSignatureForPermission>> | undefined;
 
-    const companionHasPermission = await this.companionHasPermission(position, DCAPermission.WITHDRAW);
+    if (!companionHasPermission && (useProtocolToken || hasYield)) {
+      const companionAddress = await this.contractService.getHUBCompanionAddress(LATEST_VERSION);
 
-    if (companionHasPermission) {
-      return this.meanApiService.withdrawSwappedUsingOtherToken(
```

Comment:
Don't you want to delete code from `this.meanApiService.withdrawSwappedUsingOtherToken`?